pallets/asset-registry/src/lib.rs

```diff
@@ -119,12 +123,10 @@ pub mod pallet {
 		) -> DispatchResult {
 			T::ReserveAssetModifierOrigin::ensure_origin(origin)?;
 
-			// verify asset is registered
-			let asset_multi_location =
-				AssetIdMultiLocation::<T>::get(asset_id).ok_or(Error::<T>::AssetIsNotRegistered)?;
-
 			// unregister asset
-			AssetIdMultiLocation::<T>::remove(asset_id);
+			let asset_multi_location =
+				AssetIdMultiLocation::<T>::mutate_exists(asset_id, Option::take)
+					.ok_or(Error::<T>::AssetIsNotRegistered)?;
 			AssetMultiLocationId::<T>::remove(&asset_multi_location);
```

Comment:
is it possible that you forgot to remove this one taking into account that `mutate_exists(id, Option::take)` is already removing it?

Reply:
No, there are 2 storages `AssetIdMultiLocation` and `AssetMultiLocationId`. The first is a map `id => MultiLocation` and the second is `MultiLocation => id`. The logic needs to get `MultiLocation` out of the 1st storage while removing it, and remove this `MultiLocation` from the second storage afterwards. So the second remove is just remove.
Initially, I implemented it as a functor:
```
pub fn unregister_reserve_asset(
	origin: OriginFor<T>,
	asset_id: AssetIdOf<T>,
) -> DispatchResult {
	T::ReserveAssetModifierOrigin::ensure_origin(origin)?;

	// unregister asset
	AssetIdMultiLocation::<T>::mutate_exists(asset_id, Option::take)
		.map(|asset_multi_location| {
			AssetMultiLocationId::<T>::remove(&asset_multi_location);
			Self::deposit_event(Event::ReserveAssetUnregistered {
				asset_id,
				asset_multi_location,
			});
		})
		.ok_or(Error::<T>::AssetIsNotRegistered.into())
}
```
But thought that the readability is even worse. And nesting grows. But might be an option if you like it. Or I can return back to how it was. Mainly is a question of taste.

Reply:
oh! i didn't noticed the difference about those two .. i read them very quickly and i thought you forgot to remove that. In terms of readability i agree with you that it does not add value honestly, but i think this is a very good example for an alternative removal than `get` + `remove`. So my vote goes for keeping it but adding a comment specifying exactly what you have described.

Reply:
done